build_modules/lib/src/scratch_space.dart

```diff
@@ -92,8 +94,16 @@ String _scratchSpacePackageConfig(String rootConfig) {
   var packages =
       (parsedRootConfig['packages'] as List).cast<Map<String, dynamic>>();
   for (var package in packages) {
-    package['rootUri'] = '../packages/${package['name']}';
-    package['packageUri'] = '';
+    var rootUri = packageConfigUri.resolve(package['rootUri'] as String);
+    if (rootUri == _currentDirUri) {
```

Comment:
Would it make sense to track whether we saw any package with this root URI? If it's missing does that indicate a problem?

Reply:
It does indicate a problem yes, we should see exactly one. Multiple would cause other explicit failures (you can't have identical root uris for mutliple packages).

I could add something in but probably wouldn't put it above a warning.

Reply:
Warning SGTM. My thinking is that if something goes wrong a warning here will be more clear than a strange failure in the compiler.

Reply:
warning added